contexts/credentials/v2

```diff
@@ -87,6 +87,69 @@
           "@container": "@graph"
         }
       }
+    },
+
+    "DataIntegrityProof": {
+      "@id": "https://w3id.org/security#DataIntegrityProof",
+      "@context": {
+        "@protected": true,
+        "id": "@id",
+        "type": "@type",
+        "challenge": "https://w3id.org/security#challenge",
+        "created": {
+          "@id": "http://purl.org/dc/terms/created",
+          "@type": "http://www.w3.org/2001/XMLSchema#dateTime"
+        },
+        "domain": "https://w3id.org/security#domain",
+        "expires": {
+          "@id": "https://w3id.org/security#expiration",
+          "@type": "http://www.w3.org/2001/XMLSchema#dateTime"
+        },
+        "nonce": "https://w3id.org/security#nonce",
```

Comment:
why we have this and challenge?... prefer 1.

Reply:
They have different semantics (challenge is never chosen by the signer and nonce usually is). That being said, I prefer just having `challenge` and moving any use of nonces that is specific to a particular type of crypto (e.g., BBS) into the `proofValue` via some internal encoding, just like any other crypto / proof-specific parameters or values should be.

In other words, `proofValue` should be self-contained and opaque to applications and only proof verifiers would decode it and break apart its constituent parts and do whatever is needed. This is comparable to a `jws` (that internally has parts like header and payload, etc.). That is where a signer-provided nonce belongs, IMO.

The only reason to expose any values via directly accessible `proof` properties (i.e., not encoded within `proofValue`) is if applications clearly have a need for them. Otherwise it just generates more churn for people writing generalized / layered code. If such a use can be argued for `nonce` then there's a reason to have it -- otherwise I don't think it's necessary.

Reply:
Yeah, I almost removed `nonce` but then thought that folks might complain. `nonce` is currently on the chopping block, we need to see if people see a need to keep it around.

Reply:
@tplooker had some concerns around removing `nonce` -- thoughts, @tplooker ?